src/components/TodoItem.tsx

```diff
@@ -0,0 +1,105 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+/* eslint-disable */
+import classNames from 'classnames';
+import { Todo } from '../types/Todo';
+import { useState } from 'react';
+
+type Props = {
+  todo: Todo;
+  onSave: (todoId: number, newTitle: string, completed: boolean) => void;
+  toggleTodoCompletion: (todoId: number) => void;
+  deleteSingleTodo: (todoId: number) => void;
+  loadingTodoIds: number[];
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  toggleTodoCompletion,
+  onSave,
+  deleteSingleTodo,
+  loadingTodoIds,
+}) => {
+  const { title, id, completed } = todo;
+  const [editing, setEditing] = useState(false);
+  const [newTitle, setNewTitle] = useState(title);
+
+  const handleDoubleClick = () => {
+    setEditing(true);
+  };
+
+  const handleSave = () => {
+    if (newTitle.trim() === '') {
+      deleteSingleTodo(todo.id);
+    } else
+      if (newTitle.trim() !== title.trim()) {
+        onSave(id, newTitle, completed);
+    }
+    setEditing(false);
+  };
+
+  return (
+    <section className="todoapp__main" data-cy="TodoList">
+      {/* This is a completed todo */}
+      {editing ? (
+        <input
+          type="text"
+          className="todo__title-field"
+          value={newTitle}
+          onChange={e => setNewTitle(e.target.value)}
+          onBlur={handleSave}
+          onKeyUp={e => {
+            if (e.key === 'Enter') {
+              handleSave();
+            } else if (e.key === 'Escape') {
+              setEditing(false);
+              setNewTitle(title);
+            }
+          }}
```

Comment:
Move to separate function  (transfer the function to a variable, and here only write the name of the function)